### ISSUE TITLE
How to compile only CSS/SASS files

### ISSUE THREAD

**Issue by kirubasankars** [ember-cli/ember-cli#3052]
Now its taking 9 - 11 sec to build on every file change.

I did try without antivirus enabled/ windows search index enabled.

**Comment by stefanpenner** [comment]
please provide more information.

**Comment by kirubasankars** [comment]
I am new to ember-cli. started using it from last week. 

Working with default application template given by ember cli, so live-reload enabled by default.  Running "ember serve" on cmd.

Whenever I change my css file on style folder, its taking 9- 11 sec to refresh. is this expected or I'm missing something?

**Comment by stefanpenner** [comment]
which OS are you on?
which ember-cli version?
which node version?
do you have SSD or HDD?

**Comment by kirubasankars** [comment]
which OS are you on?
Windows 8
which ember-cli version?
0.1.5
which node version?
0.10.32
do you have SSD or HDD?
SSD

Output of ember-cli attached below.
Slowest Trees                  | Total
-------------------------------+----------------
TreeMerger (stylesAndVendor)   | 495ms
TreeMerger (ExternalTree)      | 465ms
TreeMerger (appAndDependencies) | 442ms
CustomStaticCompiler           | 301ms
Class                          | 271ms

file changed styles\app.css

Build successful - 8587ms.

Slowest Trees                  | Total
-------------------------------+----------------
TreeMerger (appAndDependencies) | 1494ms
TreeMerger (ExternalTree)      | 1187ms
TreeMerger (stylesAndVendor)   | 1148ms
CustomStaticCompiler           | 941ms

file changed styles\app.css

Build successful - 8400ms.

Slowest Trees                  | Total
-------------------------------+----------------
TreeMerger (appAndDependencies) | 1435ms
TreeMerger (ExternalTree)      | 1200ms
TreeMerger (stylesAndVendor)   | 1150ms
CustomStaticCompiler           | 848ms

file changed styles\app.css

Build successful - 9036ms.

Slowest Trees                  | Total
-------------------------------+----------------
TreeMerger (appAndDependencies) | 1578ms
TreeMerger (ExternalTree)      | 1393ms
TreeMerger (stylesAndVendor)   | 1160ms
CustomStaticCompiler           | 833ms
CustomStaticCompiler           | 547ms

**Comment by stefanpenner** [comment]
seems quite slow, did you enable http://www.ember-cli.com/#symlinks-on-windows ? Hopefully sometime soon we wont need you to enable this yourself.

**Comment by kirubasankars** [comment]
After good 30 min debugging, I found tmp folder has lot of caching folders and read by build system. is it needed?

tree_merget_tmp_dest_dir_xxxxxxxx.tmp

**Comment by stefanpenner** [comment]
seems like something isn't cleaning up stuff correctly, i was noticing this earlier today as well.

**Comment by kirubasankars** [comment]
nice, you could able reproduce it. 

back to my original question. is there any way I can configure ember-cli to build only css files.

**Comment by stefanpenner** [comment]
nope, not currently. Although @joliss is starting work to make broccoli make this choice correctly for you

**Comment by kirubasankars** [comment]
Awesome. waiting for that fix.

by the way, 
live reload able to replace style tag when on css file change. broccoli can respect that.
And template/View should reload on the fly when .hbs file change without refreshing whole page. 

POC here.
http://emberjs.jsbin.com/vibut/3/edit?html,css,js,output

**Comment by stefanpenner** [comment]
> 1) live reload able to replace style tag on css file change without refreshing whole page. broccoli can respect that by just copying to dist folder and pass file name to live reload.

@aexmachina is slowly working on this.



> 2) template/View can reload on the fly when .hbs file change without refreshing whole page.

interesting, will require more work but likely a continuation of the above.

**Comment by aexmachina** [comment]
That work is done, but @rjackson's original `detect-changes` branch has failing slow tests that I can't get passing.

**Comment by kirubasankars** [comment]
This windows only issue looks like. working great in Ubuntu VM.

**Comment by stefanpenner** [comment]
seems quite slow, did you enable http://www.ember-cli.com/#symlinks-on-windows ?

**Comment by stefanpenner** [comment]
http://www.ember-cli.com/#windows has an improved tool from microsoft to help.

**Comment by xomaczar** [comment]
@kirubasankars I wrote a little ember-cli styles reloaded addon, which would refresh only your css styles and not the whole app. It supports css/scss/sass/less/styl.
https://www.npmjs.com/package/ember-cli-styles-reloader

**Comment by stefanpenner** [comment]
@xomaczar really handy, I would be interested in accepting a PR to ember-cli that does this by default.

**Comment by xomaczar** [comment]
@stefanpenner kind of brute force, if any files with the above ext change, I do http.get(liveReloadHostname + '/changed?files=*.css')
you can take a look at https://github.com/xomaczar/ember-cli-styles-reloader/blob/master/index.js

**Comment by xomaczar** [comment]
@stefanpenner I am out to merging this into ember-cli

**Comment by stefanpenner** [comment]
@xomaczar yup

**Comment by aexmachina** [comment]
Nice one, this is a really simple solution. I think this belongs in ember-cli too.

**Comment by msalahz** [comment]
:+1:

**Comment by blimmer** [comment]
:+1:  on having this in ember-cli. Right now, my sass builds takes > 20 seconds on each save. It would be nice to only incur that delay when I change styles.

**Comment by stefanpenner** [comment]
the proposed solution here doesn't resolve your issue. The solution is known, we are pending broccoli API changes to land :(.

TL;DR some joints in the build are prone to invalidation. In this case causing sass to be rebuilt for no-good reason. This is a bug, in your described scenario Sass should be near 0 time
